src/libkernel/sscp/ptx/shuffle.cpp

```diff
@@ -0,0 +1,205 @@
+                                                  __acpp_uint32 delta){
+    int tmp[2];
+    __builtin_memcpy(tmp, &value, sizeof(tmp));
+    tmp[0] = __acpp_sscp_sub_group_shr_i32(tmp[0], delta);
+    tmp[1] = __acpp_sscp_sub_group_shr_i32(tmp[1], delta);
+    __acpp_int64 result = (static_cast<__acpp_int64>(tmp[1]) << 32ull) | (static_cast<__acpp_uint32>(tmp[0]));
+    return result;
+                                                  }
+
+
+
+HIPSYCL_SSCP_CONVERGENT_BUILTIN
+__acpp_int8 __acpp_sscp_sub_group_permute_i8(__acpp_int8 value,
+                                                   __acpp_int32 mask){
+    return __acpp_sscp_sub_group_permute_i32(value, mask);
+                                                  }
+
+HIPSYCL_SSCP_CONVERGENT_BUILTIN
+__acpp_int16 __acpp_sscp_sub_group_permute_i16(__acpp_int16 value,
+                                                     __acpp_int32 mask){
+    return __acpp_sscp_sub_group_permute_i32(value, mask);
+                                                  }
+
+HIPSYCL_SSCP_CONVERGENT_BUILTIN
+__acpp_int32 __acpp_sscp_sub_group_permute_i32(__acpp_int32 value,
+                                                     __acpp_int32 mask){
+    // __acpp_uint32 active_thread_mask = get_active_mask();
+    return __nvvm_shfl_sync_bfly_i32(FULL_MASk,value, mask, 0x1f);
+                                                  }
+
+HIPSYCL_SSCP_CONVERGENT_BUILTIN
+__acpp_int64 __acpp_sscp_sub_group_permute_i64(__acpp_int64 value,
+                                                     __acpp_int32 mask){
+    int tmp[2];
+    __builtin_memcpy(tmp, &value, sizeof(tmp));
+    tmp[0] = __acpp_sscp_sub_group_permute_i32(tmp[0], mask);
+    tmp[1] = __acpp_sscp_sub_group_permute_i32(tmp[1], mask);
+    __acpp_int64 result = (static_cast<__acpp_int64>(tmp[1]) << 32ull) | (static_cast<__acpp_uint32>(tmp[0]));
+    return result;
+                                                  }
+
+
+
+HIPSYCL_SSCP_CONVERGENT_BUILTIN
+__acpp_int8 __acpp_sscp_sub_group_select_i8(__acpp_int8 value,
+                                                   __acpp_int32 id){
+    return __acpp_sscp_sub_group_select_i32(value, id);
+                                                  }
+
+HIPSYCL_SSCP_CONVERGENT_BUILTIN
+__acpp_int16 __acpp_sscp_sub_group_select_i16(__acpp_int16 value,
+                                                     __acpp_int32 id){
+    return __acpp_sscp_sub_group_select_i32(value, id);
+                                                  }
+
+HIPSYCL_SSCP_CONVERGENT_BUILTIN
+__acpp_int32 __acpp_sscp_sub_group_select_i32(__acpp_int32 value,
+                                                     __acpp_int32 id){
+    //int max_subgroup_size = __acpp_sscp_get_subgroup_max_size();
+    // int index = id%max_subgroup_size;
+    // return __builtin_amdgcn_ds_bpermute(index<<2, value);
+    // __acpp_uint32 mask = get_active_mask();
+    // This doesn't work
+    return __nvvm_shfl_sync_idx_i32(FULL_MASk, value, id, 31);
```

Comment:
What doesn't work? The code or the thing that is commented out?

Also: Typo `FULL_MASk` with lower case `k` at the end?